sql/parser/builtins.go

```diff
@@ -436,7 +436,11 @@ var builtins = map[string][]builtin{
 			types:      typeList{},
 			returnType: DummyDate,
 			fn: func(e EvalContext, args DTuple) (Datum, error) {
-				return MakeDDate(e.StmtTimestamp.Time), nil
+				loc, err := e.GetLocation()
+				if err != nil {
+					return DummyTimestamp, err
+				}
+				return MakeDDate(e.StmtTimestamp.Time.In(loc)), nil
```

Comment:
I think it would be better to set the appropriate location when we initialize `e.StmtTimestamp`. In general the server should never touch a timestamp whose location is `time.Local`. It should always be either UTC or the session location.